packages/loot-core/src/client/shared-listeners.ts

```diff
@@ -20,16 +20,30 @@ import type { Notification } from './state-types/notifications';
 import { type AppStore } from './store';
 
 export function listenForSyncEvent(store: AppStore) {
+  // TODO: Should this run on mobile too?
+  const unlistenUnauthorized = listen('sync-event', async ({ type }) => {
+    if (type === 'unauthorized') {
+      store.dispatch(
+        addNotification({
+          type: 'warning',
+          message: 'Unable to authenticate with server',
+          sticky: true,
+          id: 'auth-issue',
+        }),
+      );
+    }
+  });
+
   let attemptedSyncRepair = false;
 
-  listen('sync-event', event => {
+  const unlistenSuccess = listen('sync-event', event => {
     const prefs = store.getState().prefs.local;
     if (!prefs || !prefs.id) {
       // Do nothing if no budget is loaded
       return;
     }
 
-    if (event.type === 'success') {
+    if (event.type === 'success' || event.type === 'applied') {
```

Comment:
Refreshes the state whenever a server sync was made (i.e. `success` type) or a local change was made/applied (i.e. `applied` type)